cli/src/commands/extensions/mod.rs

```diff
@@ -106,6 +120,52 @@ async fn handle_add_extension(path: &str) -> CommandResult {
     let extension_path = PathBuf::from(path);
     let extension = Extension::try_from(extension_path)?;
 
+    let permissions = extension.permissions();
+
+    // Attempt to construct a `PermissionsOptions` from the `Permissions`
+    // object in order to validate the permissions.
+    let _ = PermissionsOptions::try_from(permissions)?;
+
+    if !accept_permissions && extension.requires_permissions() {
+        println!("The `{}` extension requires the following permissions:", extension.name());
+
+        if let Some(read_paths) = permissions.read() {
+            println!("  Read from the following paths:");
+            for path in read_paths {
+                println!("    {}", path);
+            }
+        }
```

Comment:
I'm sure this is one of the reasons why the PR is still a draft, but I'm confident we can reduce the repetition here.